integration-tests/load-tests/load-test-helpers.ts

```diff
@@ -0,0 +1,226 @@
+import { ethers } from 'ethers'
+import { Contract } from '@ethersproject/contracts'
+import { OptimismEnv } from '../test/shared/env'
+import { Direction } from '../test/shared/watcher-utils'
+
+interface TransactionParams {
+  contract: Contract
+  functionName: string
+  functionParams: any[]
+}
+
+export const executeL1ToL2Transactions = async (
+  env: OptimismEnv,
+  txs: TransactionParams[]
+) => {
+  for (const tx of txs) {
+    const signer = ethers.Wallet.createRandom().connect(env.l1Wallet.provider)
+    const receipt = await env.l1Messenger
+      .connect(signer)
+      .sendMessage(
+        tx.contract.address,
+        tx.contract.interface.encodeFunctionData(
+          tx.functionName,
+          tx.functionParams
+        ),
+        8000000,
+        {
+          gasPrice: 0,
+        }
+      )
+
+    await env.waitForXDomainTransaction(receipt, Direction.L1ToL2)
+  }
+}
+
+export const executeL2ToL1Transactions = async (
+  env: OptimismEnv,
+  txs: TransactionParams[]
+) => {
+  for (const tx of txs) {
+    const signer = ethers.Wallet.createRandom().connect(env.l2Wallet.provider)
+    const receipt = await env.l2Messenger
+      .connect(signer)
+      .sendMessage(
+        tx.contract.address,
+        tx.contract.interface.encodeFunctionData(
+          tx.functionName,
+          tx.functionParams
+        ),
+        8000000,
+        {
+          gasPrice: 0,
+        }
+      )
+
+    await env.waitForXDomainTransaction(receipt, Direction.L2ToL1)
+  }
+}
+
+export const executeL2Transactions = async (
+  env: OptimismEnv,
+  txs: TransactionParams[]
+) => {
+  for (const tx of txs) {
+    const signer = ethers.Wallet.createRandom().connect(env.l2Wallet.provider)
+    const result = await signer.sendTransaction({
+      to: tx.contract.address,
+      data: tx.contract.interface.encodeFunctionData(
+        tx.functionName,
+        tx.functionParams
+      ),
+      gasPrice: 0,
```

Comment:
I think something like `tx.contract.connect(signer).methods[tx.functionName](...tx.functionParams)` would also work, saves you from having to do the encoding yourself.